PR: https://github.com/odoo/odoo/pull/33839

From: 83aaea9e5ce92530875e4fca606a89fe423803ef
From: Kevin Baptiste
Date: 2019-06-26 06:56:17

Structural Changes: 3
Total Changes: 13

[IMP] base: Add option on image widget to enable zoom

Purpose
=======

As the employee picture on a form view is often too small,
it should be possible to zoom on it like for products in eCommerce.

Specification
=============

Added new bool options on FieldBinaryImage: `zoom` and `background`.

`zoom` will call zoomOdoo.js and display a popup with the zoomed out
image.

`background` is used to either display the image as a div element with
a background-url (to be used e.g. in a kanban view) or as a standard
img element.

closes odoo/odoo#33839

Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/hr_employee.py
+++ b/addons/hr/models/hr_employee.py
@@ -92,18 +92,11 @@ class HrEmployeePrivate(models.Model):
 
     # image: all image fields are base64 encoded and PIL-supported
     image = fields.Binary(
-        "Photo", default=_default_image,
-        help="This field holds the image used as photo for the employee, limited to 1024x1024px.")
+        "Photo", default=_default_image)
     image_medium = fields.Binary(
-        "Medium-sized photo",
-        help="Medium-sized photo of the employee. It is automatically "
-             "resized as a 128x128px image, with aspect ratio preserved. "
-             "Use this field in form views or some kanban views.")
+        "Medium-sized photo")
     image_small = fields.Binary(
-        "Small-sized photo",
-        help="Small-sized photo of the employee. It is automatically "
-             "resized as a 64x64px image, with aspect ratio preserved. "
-             "Use this field anywhere a small image is required.")
+        "Small-sized photo")
     phone = fields.Char(related='address_home_id.phone', related_sudo=False, string="Private Phone", groups="hr.group_hr_user")
     # employee in company
     parent_id = fields.Many2one('hr.employee', 'Manager')
